lib/semian/net_http.rb

```diff
@@ -67,9 +76,11 @@ def reset_exceptions
     Semian::NetHTTP.reset_exceptions
 
     def raw_semian_options
-      options = semian_configuration
-      options = options.dup unless options.nil?
-      options
+      check_semian_configuration_changed
+      @raw_semian_options ||= begin
+        @raw_semian_options = Semian::NetHTTP.retrieve_semian_configuration(address, port)
+        @raw_semian_options = @raw_semian_options.dup unless @raw_semian_options.nil?
+      end
```

Comment:
(This issue was discussed IRL, it was decided that we make `semian_configuration` only allow initialization once. Otherwise checking whether the `proc` was return different values would either (1) have a `Hash` check overhead, not good, or (2) inconsistent behaviour between assigning a `proc` that gives different values at runtime and booting different workers)